crates/sargon/src/signing/authentication/authentication_signer.rs

```diff
@@ -0,0 +1,147 @@
+use crate::prelude::*;
+
+pub struct AuthenticationSigner {
+    input: AuthenticationSigningInput,
+    interactor: Arc<dyn AuthenticationSigningInteractor>,
+}
+
+impl AuthenticationSigner {
+
+    pub fn new(
+        interactor: Arc<dyn AuthenticationSigningInteractor>,
+        profile: &Profile,
+        address_of_entity: AddressOfAccountOrPersona,
+        challenge: DappToWalletInteractionAuthChallengeNonce,
+        metadata: DappToWalletInteractionMetadata,
+    ) -> Result<Self> {
+        let input = AuthenticationSigningInput::try_from_profile(
+            profile,
+            address_of_entity,
+            challenge,
+            metadata
+        )?;
+
+        Ok(Self {
+            input,
+            interactor
+        })
+    }
+
+    pub async fn sign(self) -> Result<WalletToDappInteractionAuthProof> {
+        self.interactor.sign(
+            self.input.clone().into()
+        ).await.map(|r| WalletToDappInteractionAuthProof::from(r))
```

Comment:
If you call clippy --fix 

I think it will change this to just:

`.map(WalletTo..Proof::from)`